app/services/cocktail_service.rb

```diff
@@ -9,11 +9,18 @@ def self.random_cocktail
     parse_json(response)
   end
 
-  def self.search_cocktails(query)
+  def self.search_by_name(query)
     response = conn.get('search.php?', { s: query })
     parse_json(response)
   end
 
+  def self.search_by_ingredient(ingredient)
+    response = conn.get('filter.php?', { i: ingredient })
+    return parse_json(response) if response.body.present?
+
+    { drinks: nil }
+  end
```

Comment:
For some reason, all the other endpoints return { "drinks": null } by default if no results are found, but this endpoint just returns nothing at all, so I tried to mimic that for consistency.